demo-app/src/app/app.component.ts

```diff
@@ -1,16 +1,41 @@
-import { Component, OnInit } from "@angular/core";
+import { ChangeDetectorRef, Component, OnDestroy, OnInit, ViewChild } from "@angular/core";
+import { NavigationEnd, Router, Event } from "@angular/router";
+import { MatSidenav } from "@angular/material/sidenav";
+import { MediaMatcher } from "@angular/cdk/layout";
+import { Subscription } from "rxjs";
+
+const MEDIA_MATCH: string = "(max-width: 600px)";
 
 @Component({
 	selector: "app-root",
 	templateUrl: "./app.component.html",
 	styleUrls: ["./app.component.scss"]
 })
-export class AppComponent implements OnInit {
-	public constructor() {
-		/*empty*/
+export class AppComponent implements OnInit, OnDestroy {
+	@ViewChild("sidenav")
+	private _sidenav: MatSidenav;
+	public mobileQuery: MediaQueryList;
+
+	private _routerSubscription: Subscription;
+	private readonly _mobileQueryListener = () => null;
+
+	public constructor(changeDetectorRef: ChangeDetectorRef, media: MediaMatcher, private _router: Router) {
+		this.mobileQuery = media.matchMedia(MEDIA_MATCH);
+		this._mobileQueryListener = () => changeDetectorRef.detectChanges();
+		this.mobileQuery.addEventListener("change", this._mobileQueryListener);
 	}
 
 	public ngOnInit(): void {
-		/*empty*/
+		this._routerSubscription = this._router.events.subscribe((value: Event) => {
+			if (value instanceof NavigationEnd && this._sidenav.mode === "over") {
+				/* tslint:disable:no-floating-promises */
+				this._sidenav.close();
```

Comment:
Is `this._sidenav.close().then()` ok, since we never use the result?

Reply:
Well, that is also correct, but to be consistent in all our codebase, let's use observables instead. This will make it easier to improve this logic if needed ;)

Reply:
Ok, I updated the PR.